src/lang/std/engineConnection.ts

```diff
@@ -1491,6 +1525,38 @@ export class EngineCommandManager {
     }
     return promise
   }
+  async handlePendingBatchCommand(
+    id: string,
+    commands: Models['ModelingCmdReq_type'][],
+    ast?: Program,
+    range?: SourceRange
+  ) {
+    let resolve: (val: any) => void = () => {}
+    const promise = new Promise((_resolve, reject) => {
+      resolve = _resolve
+    })
+
+    // Add the overall batch command to the artifact map just so we can track all of the
+    // individual commands that are part of the batch.
+    // we'll delete this artifact once all of the individual commands have been processed.
+    this.artifactMap[id] = {
+      range: range || [0, 0],
+      pathToNode: [],
+      type: 'pending',
+      commandType: 'batch',
+      additionalData: { type: 'batch-ids', ids: commands.map((c) => c.cmd_id) },
+      parentId: undefined,
+      promise,
+      resolve,
+    }
+
+    await Promise.all(
+      commands.map((c) =>
+        this.handlePendingCommand(c.cmd_id, c.cmd, ast, range)
```

Comment:
This is wrong, because I'm passing in the same range for each command.

maybe we could change `sendModelingCommand` (or the wasm version)  to pass in a id-to-range map so that we can link them back up? and this line could become something like?

```suggestion
        this.handlePendingCommand(c.cmd_id, c.cmd, ast, cmdIdToRange[c.cmd_id])
```

Reply:
Or maybe even we make a `sendModelingCommands` (plural) and when it's called from wasm it's more like 
```typescript
[
  {
    cmd: {
      //...
    },
    sourceRange: [25,225], // pathToNode would be even better
  },
  {
    // more of the same
  }
]
``` 
and it's in typescript land that we map over them to grab the commands and send all of the ones that come together off as a batch?

Reply:
just did the basic map for now